pkg/user/repository.go

```diff
@@ -41,6 +35,10 @@ func (r repository) create(u *model.User) error {
 func (r repository) findByEmail(email string) (*model.User, error) {
 	var u *model.User
 	err := r.db.Where("email = ?", email).First(&u).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		err := fmt.Errorf("failed to find user with email %q: %v", email, err)
```

Comment:
I don't think we need to put the `err` into the error we construct. We know we could not find the `email`. That should be enough info to return.